### ISSUE TITLE
Context.Wait different behavior

### ISSUE THREAD

**Issue by MCKRUZ** [Microsoft/BotBuilder#1475]
So I have two different ways I call my Greeting dialog. One is through the usual method.

```
public async Task<HttpResponseMessage> Post([FromBody]Activity activity)
    {
        if (activity.Type == ActivityTypes.Message)
        {
            await Conversation.SendAsync(activity, () => new GreetingDialog());
        }
        else
        {
            HandleSystemMessage(activity);
        }
        var response = Request.CreateResponse(HttpStatusCode.OK);
        return response;
    }
```

Using this method when I get into my dialog the "messageReceivedAsync" method gets called right away after the StartAsync Method

```
        public async Task StartAsync(IDialogContext context)
    {
        await context.PostAsync("Hi I'm John Bot");
        context.Wait(MessageReceivedAsync);
    }

```
However, if I now call the same dialog in a chain like so

```
public static readonly IDialog<string> dialog = Chain.PostToChain()
        .Select(msg => msg.Text)
        .Switch(
             new RegexCase<IDialog<string>>(new Regex("^hi", RegexOptions.IgnoreCase), (context, txt) =>
             {
                return Chain.ContinueWith(new GreetingDialog(), AfterDialogContinuation);
             }),
             new DefaultCase<string, IDialog<string>>((context, txt) =>
             {
                 return Chain.ContinueWith(FormDialog.FromForm(RoomReservation.BuildForm, FormOptions.PromptInStart), AfterDialogContinuation);
             }))
         .Unwrap()
         .PostToUser();

```
It will not get called right away. Why the difference? Also is there any way to call the Greeting Dialog without having to call the AfterDialogContinuation method?

**Comment by willportnoy** [comment]
In the first version, the initial message that resulted in the HTTP POST to /api/messages executed StartAsync because there was no root dialog on the stack, send the user a message, waited for an incoming message, then satisfied that wait with the incoming message.

In the second version, it first waited for a message, then ran it through the switch/case (without the initial "send the user a message" part).  if you want to call the GreetingDialog directly, make it the first part of your chain (either as an instance of IDialog<T> or something lazilly created with Chain.From).

**Comment by alokraj68** [comment]
I guess you are waiting for a message from the user in the second one. 
you can simply send the user a message here and on his reply it will execute as your intention.

**Comment by MCKRUZ** [comment]
-Willportnoy - Are you saying that because in the first version the post essentially "was" the root dialog and it had no where else to go it satisfied the wait with MessageRecievedAsync?

- Alokraj68 - If you're talking about the question "Is there any way to call the Greting Dialog without ahving to call the AfterDialogContinuation" I don't understand your answer.  Can you provide a code example?

**Comment by willportnoy** [comment]
@MCKRUZ yes, the post started the GreetingDialog and satisfied the wait to resume at MessageReceivedAsync immediately with that first post.

Regarding your question, if you want to use the chain methods, you might use Chain.From e.g.
https://github.com/Microsoft/BotBuilder/blob/5cf71c742f27d89e9dbc4076f850122bd6edac11/CSharp/Samples/AnnotatedSandwichBot/Controllers/MessagesController.cs#L19

**Comment by willportnoy** [comment]
I'm going to close this issue, but feel free to open a new issue for further discussion.
